cloud/vmware/vmware_local_user_manager.py

```diff
@@ -0,0 +1,211 @@
+        except vmodl.RuntimeFault as runtime_fault:
+            self.module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            self.module.fail_json(msg=method_fault.msg)
+        except Exception as e:
+            self.module.fail_json(msg=str(e))
+
+
+    def check_local_user_manager_state(self):
+        user_account = self.find_user_account()
+        if not user_account:
+            return 'absent'
+        else:
+            return 'present'
+
+
+    def find_user_account(self):
+        searchStr = self.local_user_name
+        exactMatch = True
+        findUsers = True
+        findGroups = False
+        user_account = self.content.userDirectory.RetrieveUserGroups(None, searchStr, None, None, exactMatch, findUsers, findGroups)
+        return user_account
+
+
+    def create_account_spec(self):
+        account_spec = vim.host.LocalAccountManager.AccountSpecification()
+        account_spec.id = self.local_user_name
+        account_spec.password = self.local_user_password
+        account_spec.description = self.local_user_description
+        return account_spec
+
+
+    def state_create_user(self):
+        account_spec = self.create_account_spec()
+
+        try:
+            task = self.content.accountManager.CreateUser(account_spec)
+            self.module.exit_json(changed=True)
+        except vmodl.RuntimeFault as runtime_fault:
+            module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            module.fail_json(msg=method_fault.msg)
+
+    def state_update_user(self):
+        account_spec = self.create_account_spec()
+
+        try:
+            task = self.content.accountManager.UpdateUser(account_spec)
+            self.module.exit_json(changed=True)
+        except vmodl.RuntimeFault as runtime_fault:
+            module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            module.fail_json(msg=method_fault.msg)
+
+
+    def state_remove_user(self):
+        try:
+            task = self.content.accountManager.RemoveUser(self.local_user_name)
+            self.module.exit_json(changed=True)
+        except vmodl.RuntimeFault as runtime_fault:
+            module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            module.fail_json(msg=method_fault.msg)
+
+
+    def state_exit_unchanged(self):
+        self.module.exit_json(changed=False)
+
+
+
+def main():
+
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(local_user_name=dict(required=True, type='str'),
+                              local_user_password=dict(required=False, type='str'),
+                              local_user_description=dict(required=False, type='str'),
+                              state=dict(default='present', choices=['present', 'absent'], type='str')))
```

Comment:
Documentation had required: true, though not in code

Also you have a default set, which isn't documented, and also gets rid of the need for this to be `required`